tests/test-chat-template.cpp

```diff
@@ -137,9 +137,25 @@ int main(void) {
         assert(output == expected);
     }
 
-    // test llama_chat_format_single
-    std::cout << "\n\n=== llama_chat_format_single ===\n\n";
+
+    // test llama_chat_format_single for system message
+    std::cout << "\n\n=== llama_chat_format_single (system message) ===\n\n";
     std::vector<llama_chat_msg> chat2;
+    llama_chat_msg sys_msg{"system", "You are a helpful assistant"};
+
+    auto fmt_sys = [&](std::string tmpl) {
+        auto output = llama_chat_format_single(nullptr, tmpl, chat2, sys_msg, false);
+        std::cout << "fmt_sys(" << tmpl << ")\n" << output << "\n-------------------------\n";
+        return output;
+    };
+    assert(fmt_sys("chatml") == "<|im_start|>system\nYou are a helpful assistant<|im_end|>\n");
+    assert(fmt_sys("llama2") == "[INST] You are a helpful assistant\n");
+    assert(fmt_sys("gemma")  == ""); // for gemma, system message is merged with user message
+    assert(fmt_sys("llama3") == "<|start_header_id|>system<|end_header_id|>\n\nYou are a helpful assistant<|eot_id|>");
+
+
+    // test llama_chat_format_single for user message
+    std::cout << "\n\n=== llama_chat_format_single (user message) ===\n\n";
```

Comment:
nit : prefer `printf` over `std::cout`, mainly for consistency with the rest of the code

Reply:
Yup that makes sense, I changed to `printf` : https://github.com/ggerganov/llama.cpp/pull/8657/commits/20f56a02db44f94c701a6a8fb30c056b7ed2f3a4

Will merge after CI pass